sdk/resourcemanager/Azure.ResourceManager/src/Shared/SharedExtensions.cs

```diff
@@ -15,6 +16,21 @@ namespace Azure.ResourceManager.Core
     /// </summary>
     internal static class SharedExtensions
     {
+        /// <summary>
+        /// Collects the segments in a resource identifier into a string
+        /// </summary>
+        /// <param name="resourceId">the resource identifier</param>
+        /// <param name="start">the index of segment we would like to start</param>
+        /// <returns></returns>
+        public static string GetParts(this ResourceIdentifier resourceId, int start)
```

Comment:
This API will get the sub-path after `/providers`. We should give a more specific name.